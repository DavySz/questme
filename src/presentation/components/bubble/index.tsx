import { BubbleFill } from "./fill";
import { BubbleRoot } from "./root";
import { BubbleOutline } from "./outline";

export const Bubble = {
  Root: BubbleRoot,
  Fill: BubbleFill,
  Outline: BubbleOutline,
};
