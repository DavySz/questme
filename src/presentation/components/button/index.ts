import { ButtonIcon } from "./icon";
import { ButtonRoot } from "./root";
import { ButtonText } from "./text";

export const Button = {
  Root: ButtonRoot,
  Text: ButtonText,
  Icon: ButtonIcon,
};
