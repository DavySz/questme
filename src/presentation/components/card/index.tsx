import { CardGroup } from "./group";
import { CardIcon } from "./icon";
import { CardRoot } from "./root";

export const Card = {
  Root: CardRoot,
  Icon: CardIcon,
  Group: CardGroup,
};
