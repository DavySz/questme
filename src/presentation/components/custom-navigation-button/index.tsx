import { CustomNavigationIcon } from "./icon";
import { CustomNavigationButtonRoot } from "./root";

export const CustomNavigationButton = {
  Root: CustomNavigationButtonRoot,
  Icon: CustomNavigationIcon,
};
