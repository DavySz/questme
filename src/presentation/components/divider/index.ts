import { DividerRoot } from "./root";
import { DividerText } from "./text";

export const Divider = {
  Root: DividerRoot,
  Text: DividerText,
};
