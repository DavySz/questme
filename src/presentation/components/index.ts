export * from "./button";
export * from "./text";
export * from "./screen-template";
export * from "./divider";
