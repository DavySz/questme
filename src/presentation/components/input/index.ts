import { InputIcon } from "./icon";
import { InputLabel } from "./label";
import { InputRoot } from "./root";
import { InputText } from "./text";

export const Input = {
  Root: InputRoot,
  Label: InputLabel,
  Text: InputText,
  Icon: InputIcon,
};
