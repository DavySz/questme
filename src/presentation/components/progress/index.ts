import { ProgressBar } from "./bar";
import { ProgressLabel } from "./label";
import { ProgressRoot } from "./root";

export const Progress = {
  Root: ProgressRoot,
  Label: ProgressLabel,
  Bar: ProgressBar,
};
