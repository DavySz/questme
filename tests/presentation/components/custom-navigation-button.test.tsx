import { CustomNavigationButton } from "@presentation/components";
import { customRender } from "../../utils";

import PlusIcon from "@presentation/assets/icons/plus-icon.svg";

describe("CustomNavigationButton", () => {
  test("Should render component without errors", () => {
    const tree = customRender(
      <CustomNavigationButton.Root>
        <CustomNavigationButton.Icon icon={PlusIcon} />
      </CustomNavigationButton.Root>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("Should render correct icon when color is passed", () => {
    const { findByTestId } = customRender(
      <CustomNavigationButton.Root>
        <CustomNavigationButton.Icon icon={PlusIcon} color="neutral-black" />
      </CustomNavigationButton.Root>
    );

    expect(findByTestId("icon-with-color")).toBeTruthy();
  });
});
