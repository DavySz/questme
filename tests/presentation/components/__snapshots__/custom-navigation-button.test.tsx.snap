// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomNavigationButton Should render component without errors 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomLeftRadius": 32,
      "borderBottomRightRadius": 32,
      "borderTopLeftRadius": 32,
      "borderTopRightRadius": 32,
      "height": 64,
      "justifyContent": "center",
      "marginTop": -20,
      "width": 64,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#6A5AE0",
        "borderBottomLeftRadius": 24,
        "borderBottomRightRadius": 24,
        "borderTopLeftRadius": 24,
        "borderTopRightRadius": 24,
        "height": 48,
        "justifyContent": "center",
        "marginTop": -20,
        "opacity": 1,
        "width": 48,
      }
    }
  >
    <SvgMock
      testID="icon-without-color"
    />
  </View>
</View>
`;
